unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift

```diff
@@ -196,18 +205,31 @@ class WalletConnectServiceV2: WalletConnectServiceV2Protocol {
             return
         }
 
-        guard let toDisconnect = appsStorage.find(by: unifiedApp) else {
+        guard let toDisconnect = appsStorageV2.find(by: unifiedApp) else {
             Debugger.printFailure("Failed to find app to disconnect", critical: false)
             return
         }
         
-        await self.appsStorage.remove(byTopic: toDisconnect.sessionProxy.topic)
+        await self.appsStorageV2.remove(byTopic: toDisconnect.sessionProxy.topic)
         try await self.disconnect(topic: toDisconnect.sessionProxy.topic)
         self.listeners.forEach { holder in
             holder.listener?.didDisconnect(from: PushSubscriberInfo(appV2: toDisconnect))
         }
     }
     
+    private func disconnectApp(by topic: String) async throws {
+        guard let toDisconnect = appsStorageV2.find(byTopic: topic) else {
+            Debugger.printFailure("Failed to find app to disconnect", critical: false)
+            return
+        }
+        
+        await self.appsStorageV2.remove(byTopic: topic)
+        try await self.disconnect(topic: topic)
```

Comment:
What will happen if disconnect function failed and app already removed from the storage?